app/controllers/api/base_controller/parameters.rb

```diff
@@ -175,6 +177,18 @@ def sort_directive(attr, order, options)
         sort_item << " DESC" if order && order.downcase.start_with?("desc")
         sort_item
       end
+
+      def sort_virtual_attr_directive(klass, attr, order, options)
+        raise BadRequestError,
+              "#{klass.name} cannot be sorted by #{attr}" unless klass.attribute_supported_by_sql?(attr)
```

Comment:
This is great.

In reports/views, we do support sorting by non sql friendly attributes.
It is a major memory hit and very slow.